Code/Data_science_project_exploration.Rmd

```diff
@@ -0,0 +1,280 @@
+##create year/month/day/hour/minute fields
+data <- data %>% mutate(YEAR=year(DATETIME), 
+                        MONTH=month(DATETIME),
+                        DAY=day(DATETIME),
+                        HOUR=hour(DATETIME), 
+                        MINUTE=minute(DATETIME))
+
+##season
+data <- data %>%mutate(SEASON = case_when(
+  MONTH %in%  9:11 ~ "SPRING",
+  MONTH %in%  c(12, 1, 2)  ~ "SUMMER",
+  MONTH %in%  3:5  ~ "AUTUMN",
+  TRUE ~ "WINTER"))
+##create boolean for 8am~8pm timeslot
+data <- data %>%mutate(timeframe_8amto8pm = case_when(
+  HOUR %in%  8:20 ~ TRUE,
+  TRUE ~ FALSE))
+##Get days of week, monday = 1,...,sunday=7
+data<- data%>%mutate(DaysOfWeek=wday(DATETIME,
+                                     week_start = getOption("lubridate.week.start", 1)))
+##get weekday&weekend
+data <- data %>%mutate(WEEKDAYWEEKEND = case_when(
+  DaysOfWeek %in%  1:5 ~ "WEEKDAY",
+  DaysOfWeek %in%  6:7  ~ "WEEKEND",
+))
+
+##Deduplicate
+data <- data %>% distinct()
+
+```
+
+Calculate rolling datetime difference to get gaps, and summarise
+```{r Get rolling time difference}
+data <- data %>%  mutate(DATETIME_DIFF = DATETIME - lag(DATETIME))
+table(data$DATETIME_DIFF)
+
+```
+investigate the 0 min diff rows (resolved):
+```{r}
+##no more duplicates
+data %>% filter(DATETIME_DIFF==0)
+```
+
+Plot data for datetime against total demand:
+```{r plot }
+plot(data$TOTALDEMAND~data$DATETIME,type='l')
+
+```
+
+Due to the number of spikes, it is difficult to identify trends, attempting to plot by year-month, aggregating energy usage by averaging (since time interval is not consistent):
+```{r plot2}
+##drop incomplete period
+##Calculate grouped total demand and forecast
+data_YM_plot <- data %>% 
+  filter(YEARMONTH != as.Date('2022-08-01')) %>%
+  group_by(YEARMONTH) %>%
+  summarise(T_DEMAND=mean(TOTALDEMAND),
+            T_forecastfinal = mean(FINAL_FORECAST))
+
+##First row of March (Autumn)
+data_YM_plot <- data_YM_plot %>%
+  arrange(YEARMONTH) %>%
+  group_by(yr = year(YEARMONTH), mth = month(YEARMONTH)) %>%
+  # flag each first March row
+  mutate(first_Mar_row = mth ==3 & row_number()==1)
+
+##First row of June (Winter)
+data_YM_plot <- data_YM_plot %>%
+  # ensure it's sorted by date
+  arrange(YEARMONTH) %>%
+  # group by year and month
+  group_by(yr = year(YEARMONTH), mth = month(YEARMONTH)) %>%
+  # flag each first January row
+  mutate(first_Jun_row = mth ==6 & row_number()==1)
+
+##First row of September (Spring)
+data_YM_plot <- data_YM_plot %>%
+  arrange(YEARMONTH) %>%
+  group_by(yr = year(YEARMONTH), mth = month(YEARMONTH)) %>%
+  # flag each first September row
+  mutate(first_Sep_row = mth == 9 & row_number()==1)
+
+##First row of December (Summer)
+data_YM_plot <- data_YM_plot %>%
+  arrange(YEARMONTH) %>%
+  group_by(yr = year(YEARMONTH), mth = month(YEARMONTH)) %>%
+  # flag each first December row
+  mutate(first_Dec_row = mth ==12 & row_number()==1)
+
+plot(data_YM_plot$T_DEMAND~data_YM_plot$YEARMONTH,type='l')
```

Comment:
consider `ggplot` for better visualisations

Reply:
Strongly agree, ggplot is definitely nicer, will make the conversion - created those plots on a whim as thoughts come in